libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/RoadObjectsStore.kt

```diff
@@ -102,4 +161,30 @@ class RoadObjectsStore internal constructor(
     fun removeCustomRoadObject(roadObjectId: String) {
         navigator.roadObjectsStore?.removeCustomRoadObject(roadObjectId)
     }
+
+    private fun addObjectToStore(
+        roadObjectId: String,
+        locations: List<Expected<OpenLRLocation, String>>
+    ) {
+        if (locations.isEmpty()) {
+            logger.d(TAG, Message("No locations decoded for roadObjectId = $roadObjectId."))
+            return
+        }
+
+        val either = locations.first()
+        when {
+            either.isValue ->
+                either.value?.let {
+                    navigator.roadObjectsStore?.addCustomRoadObject(roadObjectId, it)
+                }
+
+            either.isError ->
+                logger.d(
+                    TAG,
+                    Message(
+                        "Decoding failed. Error = ${either.error}, roadObjectId = $roadObjectId."
+                    )
+                )
+        }
```

Comment:
Maybe?

```suggestion
        when (either.isValue){
            true ->
                navigator.roadObjectsStore?.addCustomRoadObject(roadObjectId, either.value!!)
            false ->
                logger.d(
                    TAG,
                    Message(
                        "Decoding failed. Error = ${either.error}, roadObjectId = $roadObjectId."
                    )
                )
        }
```

Reply:
changed to `if`-`else`, it's shorter than ^^